src/Psalm/Internal/Analyzer/Statements/Expression/AssertionFinder.php

```diff
@@ -28,6 +32,10 @@
 
 /**
  * @internal
+ * This class transform conditions in code into "assertions" that will be reconciled with the type already known of a
+ * given variable to narrow the type or find paradox.
+ * For example if $a is an int, if(count($a) > 0) will be turned into an assertion to make psalm understand that in the
```

Comment:
The example looks a bit off, using `count()` on an `int` is kinda unexpected (and is a TypeError in PHP 8).

Reply:
Whoops, good catch